app/src/main/java/com/babylon/wallet/android/presentation/dialogs/assets/AssetDialogViewModel.kt

```diff
@@ -207,5 +261,19 @@ class AssetDialogViewModel @Inject constructor(
                 private const val EPOCH_TIME_MINUTES = 5
             }
         }
+
+        sealed interface HideConfirmationType {
+
+            data object Asset : HideConfirmationType
+
+            data class Collection(
+                val name: String
+            ) : HideConfirmationType
```

Comment:
IMHO we get the best readability by modeling any UI data this way, even though I agree there's a degree of redundancy here now because we have the `Asset` as part of the UIState. If you agree I'd leave this as is. 

A side note (although the current case is minor and the advantages are not so noticeable): maybe long-term we could avoid referencing domain classes in the UiState directly and rather have individual properties matching the UI needs exactly.